solution/ui/regulations/test/unit-test/SupplementalContent.test.js

```diff
@@ -34,13 +32,35 @@ describe("Supplemental Content", () => {
                 subparts: ["A"],
             },
         });
+
         await flushPromises();
 
         const view = screen.getByText("Subpart A Resources");
         expect(view.id).toBe("subpart-resources-heading");
+
         const subG = screen.getByText("Subregulatory Guidance");
         expect(subG).toBeTruthy();
     });
+
+    it("Properly shows and hides empty categories", async () => {
+        render(SupplementalContent, {
+            props: {
+                apiUrl: "http://localhost:8000/",
+                title: "42",
+                part: "433",
+                subparts: ["A"],
+            },
+        });
+
+        await flushPromises();
+
+        const subEmptyButVisible = screen.queryByText("Empty but Visible");
+        expect(subEmptyButVisible).toBeTruthy();
+
+        const subEmptyAndHidden = screen.queryByText("Empty and Hidden");
```

Comment:
`queryBy...` returns `null` instead of an error, which allows us to assert against a query that doesn't find anything.  See https://testing-library.com/docs/queries/about/